File: Project/unity/Assets/MainCharacter/Script/FootSteps.js
@script RequireComponent( AudioSource )

var walkSounds:AudioClip[];

// Use this for initialization
function Start () {
}

// Update is called once per frame
function Update () 
{
	if ( transform.hasChanged && ! audio.isPlaying ) 
	{
		audio.clip = walkSounds[Random.Range(0, walkSounds.length)];
		audio.Play();
		
		transform.hasChanged = false;
		
	}
	
}

File: Project/unity/Assets/MainCharacter/Script/FootSteps.cs
using UnityEngine;
using System.Collections;

[ RequireComponent( typeof( AudioSource ) ) ]

public class FootSteps : MonoBehaviour 
{
	public GameObject target;

	public AudioClip[] walkSounds;

	// Use this for initialization
	void Start () 
	{
		target = GameObject.FindGameObjectWithTag ("Player");
	
	}
	
	// Update is called once per frame
	void Update () 
	{
		if ( target.transform.hasChanged && ! audio.isPlaying ) 
		{
			audio.clip = walkSounds[Random.Range(0, walkSounds.Length)];
			audio.Play();
			
			target.transform.hasChanged = false;
			
		}
	
	}
}
